atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/SweepQueueReader.java

```diff
@@ -15,21 +15,40 @@
  */
 package com.palantir.atlasdb.sweep.queue;
 
-import com.google.common.collect.ImmutableList;
+import java.util.Optional;
+import java.util.function.IntSupplier;
 
 class SweepQueueReader {
     private final SweepableTimestamps sweepableTimestamps;
     private final SweepableCells sweepableCells;
+    private final IntSupplier maximumPartitionsToBatchInSingleRead;
 
     SweepQueueReader(SweepableTimestamps sweepableTimestamps,
-            SweepableCells sweepableCells) {
+            SweepableCells sweepableCells,
+            IntSupplier maximumPartitionsToBatchInSingleRead) {
         this.sweepableTimestamps = sweepableTimestamps;
         this.sweepableCells = sweepableCells;
+        this.maximumPartitionsToBatchInSingleRead = maximumPartitionsToBatchInSingleRead;
     }
 
     SweepBatch getNextBatchToSweep(ShardAndStrategy shardStrategy, long lastSweptTs, long sweepTs) {
-        return sweepableTimestamps.nextSweepableTimestampPartition(shardStrategy, lastSweptTs, sweepTs)
-                .map(fine -> sweepableCells.getBatchForPartition(shardStrategy, fine, lastSweptTs, sweepTs))
-                .orElseGet(() -> SweepBatch.of(ImmutableList.of(), DedicatedRows.of(ImmutableList.of()), sweepTs - 1L));
+        SweepBatchAccumulator accumulator = new SweepBatchAccumulator(sweepTs, lastSweptTs);
+        for (int currentBatch = 0;
+                currentBatch < maximumPartitionsToBatchInSingleRead.getAsInt()
+                        && accumulator.shouldAcceptAdditionalBatch();
+                currentBatch++) {
+            Optional<Long> nextFinePartition = sweepableTimestamps.nextSweepableTimestampPartition(
+                    shardStrategy, accumulator.getProgressTimestamp(), sweepTs);
+            if (!nextFinePartition.isPresent()) {
+                return accumulator.toSweepBatch();
+            }
+            SweepBatch batch = sweepableCells.getBatchForPartition(
+                    shardStrategy, nextFinePartition.get(), accumulator.getProgressTimestamp(), sweepTs);
+            accumulator.accumulateBatch(batch);
+            if (batch.isEmpty()) {
```

Comment:
What does this do?

Reply:
discussed offline: it's probably not actually needed